### PR TITLE
SAML Assertion Encryption Tests

### PR DESCRIPTION
**Pull request by madumalt**

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLAssertionDecryptor.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        public PrivateKey getPrivateKey() {
+            return privateKey;
+        }
+
+        @Override
+        public X509Certificate getEntityCertificate() {
+            return entityCertificate;
+        }
+
+        // ********** Not implemented ************************************************************** //
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLAssertionDecryptor.java]: I don't think this comment is nice. Use standard method

File: components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLTestAssertionBuilder.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    public static Assertion buildSAMLAssertion(String issuerStr, String nameIdStr, String sessionId,
+                                               String idpEntityId, Map<String, String> userAttributeMap) {
+        DateTime now = new DateTime();
+        DateTime notOnOrAfter = now.plusMinutes(15);
+
+        Assertion samlAssertion = new AssertionBuilder().buildObject();
+
+        // Create issuer
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLTestAssertionBuilder.java]: // Create issuer.

File: components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLAssertionDecryptor.java (hunk 1 above)
Comment (madumalt) [reply components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLAssertionDecryptor.java]: This comment is to indicate that the overridden methods under the comment are not actually overridden ones. THose are just there for the sake of implementing of X509Credential interface.
AFAIK this is the conventional way of adding that kind of comments.

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java [hunk 3]
```diff
@@ -678,7 +678,7 @@ public static EncryptedAssertion setEncryptedAssertion(Assertion assertion, Stri
 
             synchronized (Runtime.getRuntime().getClass()) {
                 ssoEncrypter = (SSOEncrypter) Class.forName(IdentityUtil.getProperty(
-                        "SSOService.SAMLSSOEncrypter").trim()).newInstance();
+                        SAMLSSOConstants.SSOSERVICE_SAMLSSOENCRYPTOR_LOCATION).trim()).newInstance();
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: How about SAML_SSO_ENCRYPTOR_CONFIG_PATH ?

**Comment by mefarazath** [comment]
@madumalt there seems to be a merge conflict. Can you have a look?

**Pull request merged**